src/coffee/mappings.coffee

```diff
@@ -12,15 +12,31 @@ class CustomFieldMappings
     result = undefined
     _.each fieldDefinitions, (fieldDefinition) =>
       if fieldDefinition.name is key
-        switch fieldDefinition.type.name
-          when 'Number' then result = @mapNumber value,typeDefinitionKey,rowIndex
-          when 'Boolean' then result = @mapBoolean value,typeDefinitionKey,rowIndex
-          when 'Money' then result = @mapMoney value,typeDefinitionKey,rowIndex
-          when 'LocalizedString' then result = @mapLocalizedString value, typeDefinitionKey, rowIndex,langHeader
-          when 'Set' then result = @mapSet value,typeDefinitionKey,rowIndex,fieldDefinition.type.elementType
-          else result = value
+        # Try to map the value type by getting the appropiate function
+        try
+          if fieldDefinition.type.name == 'LocalizedString'
```

Comment:
why did is a good refactor, I think the code is more implicit and difficult to read and understand.

with the new method, we are going to be abstracting a lot of things and I think that's why we using _try/catch_.

I prefer the old method because it's easy to read and understand what's going on.